File: tests/Generator/integration/csharp/Author.cs
using System.Text.Json.Serialization;

/// <summary>
/// An simple author element with some description
/// </summary>
public class Author
{
    [JsonPropertyName("title")]
    public string Title { get; set; }
    [JsonPropertyName("email")]
    public string Email { get; set; }
    [JsonPropertyName("categories")]
    public string[] Categories { get; set; }
    [JsonPropertyName("locations")]
    public Location[] Locations { get; set; }
    [JsonPropertyName("origin")]
    public Location Origin { get; set; }
}

File: tests/Generator/resource/visualbasic/default/Author.vb
Imports System.Text.Json.Serialization

' An simple author element with some description
Public Class Author
    <JsonPropertyName("title")>
    Public Property Title As String

    <JsonPropertyName("email")>
    Public Property Email As Nullable(String)

    <JsonPropertyName("categories")>
    Public Property Categories As Nullable(String())

    <JsonPropertyName("locations")>
    Public Property Locations As Nullable(Location())

    <JsonPropertyName("origin")>
    Public Property Origin As Nullable(Location)

End Class

